presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/ValidateDependenciesChecker.java

```diff
@@ -709,10 +712,17 @@ public Void visitLateralJoin(LateralJoinNode node, Set<VariableReferenceExpressi
                     .addAll(boundVariables)
                     .build();
         }
-    }
 
-    private static void checkDependencies(Collection<VariableReferenceExpression> inputs, Collection<VariableReferenceExpression> required, String message, Object... parameters)
-    {
-        checkArgument(ImmutableSet.copyOf(inputs).containsAll(required), message, parameters);
```

Comment:
why we have this change?

Reply:
`checkDependencies` checks if referenced variables in assignment are same as input. If the assignment is type only cast (for example cast varchar(3) to varchar(5)), the referenced variable will be implicitly changed to target type. 
Reading `symbol1, VARCHAR(3)` input into `symbol1 VARCHAR(5)` still should be OK. 

But simple containAll will iterate both container and call equals method, thus won't work any more.

Reply:
got it; I assume it is needed now due to symbol to variable change? Maybe add a comment and the commit body to highlight the reason.

Reply:
Exactly, I will update the commit message.